File: scripts/inspect_db_metadata.py
import sys
import os

sys.stdout.reconfigure(encoding='utf-8')

conn_str = ''
with open('.env', 'r', encoding='utf-8') as f:
    for line in f:
        if line.startswith('DATABASE_URL=') or line.startswith('SUPABASE_POSTGRES_URL='):
            conn_str = line.split('=', 1)[1].strip().strip('"').strip("'")
            if conn_str:
                break

if '5432' in conn_str:
    conn_str = conn_str.replace('5432', '6543')

try:
    import psycopg2
    conn = psycopg2.connect(conn_str)
    cur = conn.cursor()

    cur.execute('SELECT id, code, name FROM departments ORDER BY id')
    print('=== DEPARTMENTS ===')
    for r in cur.fetchall():
        print(r)

    cur.execute('SELECT id, code, name, department_id FROM roles ORDER BY id')
    print('\n=== ROLES ===')
    for r in cur.fetchall():
        print(r)

    cur.execute('SELECT id, code, name FROM milestones ORDER BY id')
    print('\n=== MILESTONES ===')
    for r in cur.fetchall():
        print(r)

    cur.execute('SELECT id, code, name, data_type FROM properties ORDER BY id')
    print('\n=== PROPERTIES ===')
    for r in cur.fetchall():
        print(r)

    conn.close()
except Exception as e:
    print('Error:', e)

File: scripts/inspect_db_metadata.ts
import { db, roles, milestones, properties, departments } from "../src/db"

async function run() {
  const depts = await db.select().from(departments)
  console.log("=== DEPARTMENTS ===")
  for (const d of depts) {
    console.log(`${d.id} | ${d.code} | ${d.name}`)
  }

  const allRoles = await db.select().from(roles)
  console.log("\n=== ROLES (" + allRoles.length + ") ===")
  for (const r of allRoles) {
    console.log(`${r.id} | ${r.code || ""} | ${r.name} | dept:${r.departmentId} | method:${r.planningMethod}`)
  }

  const ms = await db.select().from(milestones)
  console.log("\n=== MILESTONES (" + ms.length + ") ===")
  for (const m of ms) {
    console.log(`${m.id} | ${m.code} | ${m.name}`)
  }

  const props = await db.select().from(properties)
  console.log("\n=== PROPERTIES (" + props.length + ") ===")
  for (const p of props) {
    console.log(`${p.id} | ${p.code} | ${p.name} | ${p.dataType} | unit:${p.unit}`)
  }

  process.exit(0)
}

run().catch((e) => {
  console.error(e)
  process.exit(1)
})
